### PR TITLE
[Linux] add proper desktop entry for the map editor

### LINKED ISSUE #1080: `make install` fails due to missing `vcmilauncher.desktop`
Noticed today the following install failures:
```
-- Installing: /dev/shm/portage/games-strategy/vcmi-9999/image/usr/share/vcmi/mapeditor/icons/menu-game.png
CMake Error at mapeditor/cmake_install.cmake:70 (file):
  file INSTALL cannot find
  "/dev/shm/portage/games-strategy/vcmi-9999/work/vcmi-9999/mapeditor/vcmilauncher.desktop":
  No such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:61 (include)
```

AFAIU `mapeditor` entry is unexpected:

```
$ git grep -F vcmilauncher.desktop | cat
launcher/CMakeLists.txt:                install(FILES "vcmilauncher.desktop" DESTINATION share/applications)
mapeditor/CMakeLists.txt:       install(FILES "vcmilauncher.desktop" DESTINATION share/applications)

$ find -name vcmilauncher.desktop
./launcher/vcmilauncher.desktop
```

Should `mapeditor` mention other file name? Probably added in 0b9f601d2c054da1cae85f8b3e006a623eb3f7f2 /cc @Nordsoft91

### PR DESCRIPTION
**Pull request by kambala-decapitator**
fixed #1080

### CODE REVIEW

**Review by trofi — APPROVED** [review]

**Comment by SoundSSGood** [comment]
There is an icon file in the mapeditor directory. PNG files for linux can be obtained from it. Please use them in the vcmieditor.desktop file.

mapeditor.32x32.png
mapeditor.48x48.png
mapeditor.128x128.png
mapeditor.256x256.png
mapeditor 32x32
mapeditor 48x48
mapeditor 128x128
mapeditor 256x256

**Comment by kambala-decapitator** [comment]
@SoundSSGood could you tell what needs to be done exactly? I'm not a Linux user. ideally, provide code suggestion please.

File: mapeditor/CMakeLists.txt [hunk 1]
```diff
@@ -132,5 +132,5 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
 install(DIRECTORY icons DESTINATION ${DATA_DIR}/mapeditor)
 # Install icons and desktop file on Linux
 if(NOT WIN32 AND NOT APPLE)
-	install(FILES "vcmilauncher.desktop" DESTINATION share/applications)
+	install(FILES "vcmieditor.desktop" DESTINATION share/applications)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:135]:
```suggestion
	install(FILES "${CMAKE_SOURCE_DIR}/mapeditor/icons/mapeditor.64x64.png"   DESTINATION share/icons/hicolor/64x64/apps RENAME mapeditor.png)
	install(FILES "${CMAKE_SOURCE_DIR}/mapeditor/icons/mapeditor.48x48.png"   DESTINATION share/icons/hicolor/48x48/apps RENAME mapeditor.png)
	install(FILES "${CMAKE_SOURCE_DIR}/mapeditor/icons/mapeditor.32x32.png"   DESTINATION share/icons/hicolor/32x32/apps RENAME mapeditor.png)
	install(FILES "${CMAKE_SOURCE_DIR}/mapeditor/icons/mapeditor.256x256.png" DESTINATION share/icons/hicolor/256x256/apps RENAME mapeditor.png)
	install(FILES "vcmieditor.desktop" DESTINATION share/applications)
```

File: mapeditor/vcmieditor.desktop [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Name=VCMI Map Editor
+GenericName=Strategy Game Engine
+Comment=Map editor for open engine of Heroes of Might and Magic 3
+Icon=vcmiclient
```
Comment (SoundSSGood) [inline mapeditor/vcmieditor.desktop:6]:
```suggestion
Icon=mapeditor
```

File: mapeditor/CMakeLists.txt (hunk 1 above)
Comment (SoundSSGood) [reply mapeditor/CMakeLists.txt:135]: And the last one missed file.
mapeditor.64x64.png

mapeditor 64x64

**Comment by kambala-decapitator** [comment]
@SoundSSGood thank you very much!

File: mapeditor/CMakeLists.txt [hunk 3]
```diff
@@ -132,5 +132,10 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
...
+	install(FILES "icons/mapeditor.32x32.png"   DESTINATION share/icons/hicolor/32x32/apps RENAME mapeditor.png)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:136]:
```suggestion
	install(FILES "icons/mapeditor.32x32.png"   DESTINATION share/icons/hicolor/32x32/apps RENAME vcmieditor.png)
```

File: mapeditor/CMakeLists.txt [hunk 4]
```diff
@@ -132,5 +132,10 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
...
+	install(FILES "icons/mapeditor.48x48.png"   DESTINATION share/icons/hicolor/48x48/apps RENAME mapeditor.png)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:137]:
```suggestion
	install(FILES "icons/mapeditor.48x48.png"   DESTINATION share/icons/hicolor/48x48/apps RENAME vcmieditor.png)
```

File: mapeditor/CMakeLists.txt [hunk 5]
```diff
@@ -132,5 +132,10 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
...
+	install(FILES "icons/mapeditor.64x64.png"   DESTINATION share/icons/hicolor/64x64/apps RENAME mapeditor.png)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:138]:
```suggestion
	install(FILES "icons/mapeditor.64x64.png"   DESTINATION share/icons/hicolor/64x64/apps RENAME vcmieditor.png)
```

File: mapeditor/CMakeLists.txt [hunk 6]
```diff
@@ -132,5 +132,10 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
...
+	install(FILES "icons/mapeditor.128x128.png" DESTINATION share/icons/hicolor/128x128/apps RENAME mapeditor.png)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:139]:
```suggestion
	install(FILES "icons/mapeditor.128x128.png" DESTINATION share/icons/hicolor/128x128/apps RENAME vcmieditor.png)
```

File: mapeditor/CMakeLists.txt [hunk 7]
```diff
@@ -132,5 +132,10 @@ install(TARGETS vcmieditor DESTINATION ${BIN_DIR})
...
+	install(FILES "icons/mapeditor.256x256.png" DESTINATION share/icons/hicolor/256x256/apps RENAME mapeditor.png)
```
Comment (SoundSSGood) [inline mapeditor/CMakeLists.txt:140]:
```suggestion
	install(FILES "icons/mapeditor.256x256.png" DESTINATION share/icons/hicolor/256x256/apps RENAME vcmieditor.png)
```

File: mapeditor/vcmieditor.desktop [hunk 8]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Name=VCMI Map Editor
+GenericName=Strategy Game Engine
+Comment=Map editor for open engine of Heroes of Might and Magic 3
+Icon=mapeditor
```
Comment (SoundSSGood) [inline mapeditor/vcmieditor.desktop:6]:
```suggestion
Icon=vcmieditor
```

**Review by SoundSSGood — COMMENTED** [review]
this makes sense according to the general style of VCMI naming.

**Pull request merged**